packages/core/src/agent/MessageReceiver.ts

```diff
@@ -174,12 +186,65 @@ export class MessageReceiver {
     const MessageClass = this.dispatcher.getMessageClassForType(messageType)
 
     if (!MessageClass) {
-      throw new AriesFrameworkError(`No message class found for message type "${messageType}"`)
+      throw new ProblemReportError(`No message class found for message type "${messageType}"`, {
+        problemCode: ProblemReportReason.MessageParseFailure,
+      })
     }
 
     // Cast the plain JSON object to specific instance of Message extended from AgentMessage
     const message = JsonTransformer.fromJSON(unpackedMessage.message, MessageClass)
 
     return message
   }
+
+  /**
+   * Validate an AgentMessage instance.
+   * @param message agent message to validate
+   */
+  private async validateMessage(message: AgentMessage) {
+    try {
+      await MessageValidator.validate(message)
+    } catch (error) {
+      this.logger.error(`Error validating message ${message.type}`, {
+        errors: error,
+        message: message.toJSON(),
+      })
+      throw new ProblemReportError(`No message class found for message type "${message.type}"`, {
+        problemCode: ProblemReportReason.MessageParseFailure,
+      })
+    }
+  }
+
+  /**
+   * Send the problem report message (https://didcomm.org/notification/1.0/problem-report) to the recipient.
+   * @param message error message to send
+   * @param connection connection to send the message to
+   * @param unpackedMessage received unpackedMessage
+   */
+  private async sendProblemReportMessage(
```

Comment:
I think we should add an check here whether the received message is of `@type` == `https://didcomm.org/notification/1.0/problem-report`and otherwise throw an error (not a problem report error) that we are not sending a problem report as response to a problem.  to prevent accidential problem report loops from happening. We should probably never respond with a problem report to a problem report.